### PR TITLE
Use process.env.FIREBASE_PROJECT for Firebase config values if available

### PR DESCRIPTION
**Pull request by laurenzlong**
### Description

Use FIREBASE_PROJECT and CLOUD_RUNTIME_CONFIG during population of function.config(), in anticipating of these env variables being available in the runtime in the future.

### CODE REVIEW

**Comment by laurenzlong** [comment]
Must use with this branch of the CLI: https://github.com/firebase/firebase-tools/pull/328

File: src/config.ts [hunk 1]
```diff
@@ -42,23 +42,30 @@ export namespace config {
...
+  }
+
   try {
-    loaded = require('../../../.runtimeconfig.json');
+    let path = process.env.CLOUD_RUNTIME_CONFIG || '../../../.runtimeconfig.json';
+    loadedFromFile = require(path);
   } catch (e) {
     try {
-      loaded = require('../../../config.json');
+      loadedFromFile = require('../../../config.json');
```
Comment (inlined) [inline src/config.ts]: This caused confusion with @samtstern's functions and was meant to be alpha behavior. Should we cut this or give a deprecation notice?

File: spec/config.spec.ts [hunk 2]
```diff
@@ -58,6 +58,6 @@ describe('config()', () => {
 
   it('throws an error if Firebase configs not present', () => {
     mockRequire('../../../.runtimeconfig.json', {});
-    expect(config).to.throw('Firebase config variables are missing.');
+    expect(config).to.throw('Firebase config variables are not available.');
```
Comment (inlined) [inline spec/config.spec.ts]: We added more complex behaviors. Could we test them? Hint: I'm pretty sure I see a bug =p

File: src/config.ts (hunk 1 above)
Comment (mbleigh) [inline src/config.ts]: Cut it.

**Pull request merged**